### PR TITLE
Resolve symlinks during unpacking

### LINKED ISSUE #230: Symlinks are not resolved after a submodule is cloned
Heyah. Thanks for the awesome work!

I have a cargo dependency that contains a submodule, and it doesn't seem like it gets cloned properly by (I think).

Dep's submodule: https://github.com/lambda-fairy/maud/blob/main/maud_macros/src/escape.rs
nix version: 2.5.1
`flake.nix`:

```nix
{
  description = "";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs";
    flake-utils.url = "github:numtide/flake-utils";
    naersk.url = "github:nix-community/naersk";
  };

  outputs = { self, nixpkgs, nixos-unstable, flake-utils, naersk }:
    flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
      let pkgs = nixpkgs.legacyPackages.${system};
           naersk-lib = naersk.lib."${system}".override {
             cargo = pkgs.cargo;
             rustc = pkgs.rustc;
           };
      in
      rec {
        packages.foo = naersk-lib.buildPackage {
          pname = "foo";
          root = ./.;
        };

        defaultPackage = packages.foo;

        apps.foo = flake-utils.lib.mkApp {
          drv = packages.foo;
        };

        defaultApp = apps.foo;

        devShell = pkgs.mkShell {
          nativeBuildInputs = with pkgs; [
            pkgs.rustc
            pkgs.cargo
          ];
        };
      });
}
```

`nix build` log:

```
Compiling maud_macros v0.23.0 (https://github.com/lambda-fairy/maud?branch=main#e6787cd6)
error: could not compile `maud_macros` due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
error[E0583]: file not found for module `escape`
 --> /sources/maud_macros-0.23.0-e6787cd62165a075c7f16a32f8bbacc398f52d13/src/lib.rs:9:1
  |
9 | mod escape;
  | ^^^^^^^^^^^
  |
  = help: to create the module `escape`, create file "/nix/store/shixfmbv64zzi9i0rzb9c8870ydm7cf2-crates-io/maud_macr>


error[E0425]: cannot find function `escape_to_string` in module `escape`
   --> /sources/maud_macros-0.23.0-e6787cd62165a075c7f16a32f8bbacc398f52d13/src/generate.rs:272:17
    |
272 |         escape::escape_to_string(string, &mut self.tail);
    |                 ^^^^^^^^^^^^^^^^ not found in `escape`


error: aborting due to 2 previous errors


Some errors have detailed explanations: E0425, E0583.

For more information about an error, try `rustc --explain E0425`.

[naersk] cargo returned with exit code 101, exiting
```

The strange part is that running `cargo build` directly builds it just fine. I tried looking around the issues, and it doesn't seem like any of the ones I found are exactly this problem. Any pointers are appreciated!

### PR DESCRIPTION
**Pull request by Patryk27**
Closes #230 

# Abstract

It's one of those one-byte-changes - during Git dependency unpacking, instead of:
```
cp -r $(dirname $toml) $dest
```
... what I'm proposing to do instead is:
```
cp -rL $(dirname $toml) $dest
```
... where `-L` tells `cp` to resolve symlinks - i.e. after this operations, symlinks will be substituted with the contents of the files they point at.

Overall, this will allow us to correctly compile packages such as [maud](https://github.com/lambda-fairy/maud) that contain cross-workspace-package symlinks (e.g. [this boi](https://github.com/lambda-fairy/maud/blob/d05bcba02ca4bba4e84dff50bfcfe9877c67c10d/maud_macros/src/escape.rs) that points _outside of_ `maud_macros`).

Without this fix, our new test would fail with:
```
error: couldn't read /nix/store/...-crates-io/dep/src/../symlink.txt: No such file or directory (os error 2)
 --> /sources/.../src/lib.rs:2:5
  |
  |     include_str!("../symlink.txt")
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

... and a similar error would be generated for `maud`, as can be seen in issue linked at the top.

# Potential downsides

This fix resolves _all_ symlinks - in reality, it should be enough to resolve only those symlinks that point outside of the currently processed crate; in my opinion that'd make the implementation more difficult for no apparent gain, though.

# Caveats

For clarity, this pull request depends on https://github.com/nix-community/naersk/pull/232; after that one gets merged, this PR should remain with just one commit inside of it.

### CODE REVIEW

**Comment by blitz** [comment]
@Patryk27 I've merged #232. Can you rebase this PR?

**Comment by Patryk27** [comment]
Ready! 🚀

File: test.nix [hunk 1]
```diff
@@ -170,6 +170,43 @@ let
...
+          '';
+
+          app = pkgs.runCommand "app" {
+            buildInputs = [ pkgs.git ];
+          } ''
+            mkdir $out
+            cd $out
+            cp -ar ${./test/git-symlink/app}/* .
+
+            depPath="${dep}"
```
Comment (Patryk27) [inline test.nix:196]: This part is the same as for `git-dep-by-branch-with-slash` - for simplicity, I'm planning on refactoring it (i.e. extracting to some common function) in a separate commit later, after this PR.

**Comment by Patryk27** [comment]
In the meantime I've found a much better way to approach this issue, so let me close this merge request and open a new one 🙂

**Pull request closed without merging**